### ISSUE TITLE
Rewrite as a pure-Recoil library.

### ISSUE THREAD

**Issue by jmalloc** [recoilphp/amqp#75]

**Comment by jmalloc** [comment]
@koden-km I think you should run the existing peridot test suite and check out the output, I think you'll like it.

**Comment by koden-km** [comment]
The output looks pretty nice to read. The tests look a bit more cluttered though.
Almost looks like the tests will need tests to verify they are testing correctly?

**Comment by jmalloc** [comment]
I don't really see it as any different to PHPUnit tests, though I agree they look cluttered just because of the closures.  It should make the purpose of the tests a lot clearer though, and hence help you understand how a given class is intended to work - which I suppose is the whole point :)

**Comment by koden-km** [comment]
Yeah the way they read, especially in the test output makes it clear what they are testing.
